File: Project Netwerken/src/Global/src/model/command/IDAssign.java
package model.command;

import utils.VarInt;

import java.io.ByteArrayOutputStream;
import java.io.IOException;
import java.nio.ByteBuffer;

/**
 * You will receive this data from the server.
 */
public class IDAssign extends Command {
    private static int TYPE = 2;
    private long random;
    private int id;

    public IDAssign(ByteBuffer buffer) {
        super(TYPE, buffer);
        random = VarInt.readUnsignedVarLong(buffer);
        id = VarInt.readUnsignedVarInt(buffer);
    }

    /**
     * @param random unique id of node
     * @param id index of vector clock/nodeId
     */
    public IDAssign(int nodeId, VectorClock vectorClock, long random, int id) {
        super(TYPE, nodeId, vectorClock);
        assert nodeId != 0;
        assert random >= 0;
        assert id != 0;
        this.random = random;
        this.id = id;
    }

    /**
     * @return unique id of node
     */
    public long getRandom() {
        return random;
    }

    /**
     * @return index of vector clock
     */
    public int getId() {
        return id;
    }

    @Override
    public void write(ByteArrayOutputStream out) throws IOException {
        super.write(out);
        VarInt.writeUnsignedVarLong(random, out);
        out.write(VarInt.writeUnsignedVarInt(id));
    }
}

File: Project Netwerken/src/NodeJS/global/model/command/IDAssign.js
const Command  = require('./Command');
const varint = require('../../utils/varint/varint');

class IDAssign extends Command {

    constructor(options) {
        super(options);
        if (options.buffer) {
            // constructor(buffer)
            this.random = varint.decode(options.buffer);
            this.id = varint.decode(options.buffer);
        } else {
            // constructor(nodeId, vectorClock, random, id)
            this.random = options.random;
            this.id = options.id;
        }
    }

/*    // OLD CONSTRUCTOR
    constructor(buffer) {
        super(buffer);
        this.random = varint.decode(buffer);
        this.id = varint.decode(buffer);
    }

    // OLD CONSTRUCTOR
    constructor(nodeId, vectorClock, random, id) {
        super(2, nodeId, vectorClock);
        this.random = random;
        this.id = id;
    }*/

    write(buffer) {
        super.write(buffer);
        buffer.write(varint.encode(this.random));
        buffer.write(varint.encode(this.id));
    }
}